File: api/app/model/user_model.py

from email.policy import default
from view import mongoDB as mongo

class User(mongo.Document):
    """
    Model of user

    Args:
        mongo (mongo db lib): model for the user
    """
    first_name = mongo.StringField()
    last_name = mongo.StringField() 
    email = mongo.StringField()
    password = mongo.StringField()
    role = mongo.StringField(default="user")
    device_uuid=mongo.StringField(default="")
    device_name=mongo.StringField(default="")

File: back/src/modules/users/models/User.model.js
const mongoose = require("mongoose");

const UserSchema = mongoose.Schema({
	email: { type: String, unique: true, required: true },
	firstName: { type: String },
	lastName: { type: String },
	password: { type: String, required: true },
	role: { type: String, default: "user", enum: ["admin", "user"] },
});

module.exports = mongoose.model("User", UserSchema);
